File: TreasurersApp/ClientApp/src/app/models/AddressActionResult.ts
import { Address } from './Address';
import { AddressService } from '../maintenance/address-maintenance/address.service';

export class AddressActionResult {
  success = false;
  statusMessages: string[] = [];
  address: Address = null;

  constructor(private addressService: AddressService) {
    this.success = false;
    this.address = this.addressService.newAddress();
    this.statusMessages = [];
  }
}

File: TreasurersApp/Models/AddressActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class AddressActionResult : WebResult<Address>
    {
        public AddressActionResult(bool success, List<string> statusMessages, Address address)
            :base(success, statusMessages, address)
        {
        }
    }
}
